polygeist/tools/cgeist/driver.cc

```diff
@@ -294,16 +295,23 @@ static void enableOptionsPM(mlir::PassManager &pm) {
   pm.enableTiming(timing);
 }
 
+static bool isFuncOpName(llvm::StringRef FuncOpName) {
+  return FuncOpName == mlir::func::FuncOp::getOperationName();
+}
+
 // MLIR canonicalization & cleanup.
-static int canonicalize(mlir::MLIRContext &context,
-                        mlir::OwningOpRef<mlir::ModuleOp> &module) {
-  mlir::PassManager pm(&context);
+template <typename ModuleTy>
+static int canonicalize(mlir::MLIRContext &context, ModuleTy &module,
+                        llvm::StringRef FuncOpName) {
+  mlir::PassManager pm(&context, ModuleTy::OperationT::getOperationName());
   enableOptionsPM(pm);
 
-  mlir::OpPassManager &optPM = pm.nest<mlir::func::FuncOp>();
+  mlir::OpPassManager &optPM = pm.nest(FuncOpName);
```

Comment:
`FuncOpName` is a StringRef, suggest to use an `OperationName` instead of a StringRef because the former is more specific than the latter. Is that possible ?

Reply:
No longer applies thanks to better pass manager nesting.